OSX 10.11: Undefined symbols for architecture x86_64: boost::network::http::impl::ssl_delegate::ssl_delegate()

```
[100%] Linking CXX executable ../../kovri
Undefined symbols for architecture x86_64:
  "boost::network::http::impl::ssl_delegate::ssl_delegate(boost::asio::io_service&, bool, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, long)", referenced from:
      std::__1::shared_ptr<boost::network::http::impl::ssl_delegate> std::__1::shared_ptr<boost::network::http::impl::ssl_delegate>::make_shared<boost::asio::io_service&, bool&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&, long&>(boost::asio::io_service&&&, bool&&&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&&&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&&&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&&&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&&&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&&&, boost::optional<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >&&&, long&&&) in libkovri-client.a(http.cc.o)
  "boost::network::http::impl::ssl_delegate::~ssl_delegate()", referenced from:
      std::__1::__shared_ptr_emplace<boost::network::http::impl::ssl_delegate, std::__1::allocator<boost::network::http::impl::ssl_delegate> >::~__shared_ptr_emplace() in libkovri-client.a(http.cc.o)
      std::__1::__shared_ptr_emplace<boost::network::http::impl::ssl_delegate, std::__1::allocator<boost::network::http::impl::ssl_delegate> >::~__shared_ptr_emplace() in libkovri-client.a(http.cc.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [kovri] Error 1
make[2]: *** [src/app/CMakeFiles/kovri-app.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [all-options] Error 2
```

`0.13-release` 87dd5510f479107f56764dade0454bd587cb444c
`AppleClang 7.3.0.7030031`

Complete build log here: https://build.getmonero.org/builders/kovri-tests-osx-10.11/builds/0/steps/compile/logs/stdio

Note: cpp-netlib (apparently) can't find openssl on OSX 10.11 **but** `find_package(OpenSSL)` successfully finds openssl with the same version of cmake on the same system when building kovri against openssl.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,15 @@ set(Boost_USE_MULTI_THREADED ON)
 find_package(Boost 1.58.0 REQUIRED COMPONENTS system thread)
 
 if (CPP-NETLIB_ENABLE_HTTPS)
-    find_package( OpenSSL )
+  if (APPLE)
+    # If we're on OS X check for Homebrew's copy of OpenSSL instead of Apple's
+    if (NOT OpenSSL_DIR)
+      execute_process (COMMAND brew --prefix openssl
+        OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+        OUTPUT_STRIP_TRAILING_WHITESPACE)
+    endif()
+  endif()
+  find_package(OpenSSL)
 endif()
 
 find_package( Threads )

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,15 +49,23 @@ set(Boost_USE_MULTI_THREADED ON)
 find_package(Boost 1.58.0 REQUIRED COMPONENTS system thread)
 
 if (CPP-NETLIB_ENABLE_HTTPS)
-  if (APPLE)
-    # If we're on OS X check for Homebrew's copy of OpenSSL instead of Apple's
-    if (NOT OpenSSL_DIR)
-      execute_process (COMMAND brew --prefix openssl
-        OUTPUT_VARIABLE OPENSSL_ROOT_DIR
-        OUTPUT_STRIP_TRAILING_WHITESPACE)
-    endif()
-  endif()
-  find_package(OpenSSL)
+ if (APPLE)
+   # If we're on OS X check for Homebrew's copy of OpenSSL instead of Apple's
+   if (NOT OpenSSL_DIR)
+     find_program(HOMEBREW brew)
+     if (HOMEBREW STREQUAL "HOMEBREW-NOTFOUND")
+       message(WARNING "Homebrew not found: not using Homebrew's OpenSSL")
+       if (NOT OPENSSL_ROOT_DIR)
+         message(WARNING "Use -DOPENSSL_ROOT_DIR for non-Apple OpenSSL")
+       endif()
+     else()
+       execute_process(COMMAND brew --prefix openssl
+         OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+         OUTPUT_STRIP_TRAILING_WHITESPACE)
+     endif()
+   endif()
+ endif()
+ find_package(OpenSSL)
 endif()
 
 find_package( Threads )

